presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -428,7 +429,8 @@ public PlanOptimizers(
                         estimatedExchangesCostCalculator,
                         ImmutableSet.of(
                                 new RemoveRedundantIdentityProjections(),
-                                new PushAggregationThroughOuterJoin())),
+                                new PushAggregationThroughOuterJoin(),
+                                new RemoveRedundantCrossJoin())),
```

Comment:
@Praveen2112  Any particular reason you added it here?
_Maybe_ this could go together with 
https://github.com/prestosql/presto/blob/b15ad53a359f6c8cbf91b21b453b828b0cb82782/presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java#L310-L335

@martint do you remember why `RemoveUnreferencedScalarSubqueries` isn't there?
And why we run this suite only once?

Reply:
Initially we have aaded in L#310 but expression in the scalar query gets translated to the probe side of the join after we run the `PredicatePushDown` optimizer so added it along with `RemoveRedundantIdentityProjections`

Reply:
@martint ?

Reply:
Yeah, ideally all rules should be in the same IterativeOptimizer, unless they cause the optimizer to take a greedy step it won't be able to get out of (such as how Intersect and Except are converted to a combination of joins and aggregations).

Reply:
Let's add it to both places (for future-proofing), and add a comment here explaining why it has to appear here for now.

Reply:
have added it at both the places